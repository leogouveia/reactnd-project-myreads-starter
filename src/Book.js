import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BookshelfChanger from './BookshelfChanger';

export class Book extends Component {
  state = {};

  render() {
    const { titleName, authors, coverImage } = this.props;
    return (
      <div className="book">
        <div className="book-top">
          <div
            className="book-cover"
            style={{
              width: 128,
              height: 193,
              backgroundImage: `url(${coverImage})`,
            }}
          />
          <BookshelfChanger />
        </div>
        <div className="book-title">{titleName}</div>
        <div className="book-authors">{authors.concat(' ')}</div>
      </div>
    );
  }
}

Book.propTypes = {
  titleName: PropTypes.string.isRequired,
  authors: PropTypes.array.isRequired,
  coverImage: PropTypes.string.isRequired,
};
export default Book;
