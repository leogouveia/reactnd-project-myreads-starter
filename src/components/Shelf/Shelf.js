import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Shelf.css";
export class Shelf extends Component {
  state = {};

  static defaultProps = {
    name: ""
  };

  static propTypes = {
    name: PropTypes.string.isRequired,
    children: PropTypes.node
  };

  render() {
    const { name, children } = this.props;
    return (
      <div className="bookshelf">
        <h2 className="bookshelf-title">{name}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">{children}</ol>
        </div>
      </div>
    );
  }
}

export default Shelf;
