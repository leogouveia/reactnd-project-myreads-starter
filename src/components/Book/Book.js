import React, { Component } from "react";
import PropTypes from "prop-types";
import _ from "lodash";
import "./Book.css";
import noCoverImage from "./no-cover.jpg"

export class Book extends Component {
  state = {};

  render() {
    const { book, children } = this.props;
    const { title, authors, thumbnail } = this.getBookData(book);

    return (
      <div className="book">
        <div className="book-top">
          <div
            className="book-cover"
            style={{
              width: 128,
              height: 193,
              backgroundImage: `url(${thumbnail || noCoverImage})`
            }}
          />
          {children}
        </div>
        <div className="book-title">{title}</div>
        <div className="book-authors">{authors}</div>
      </div>
    );
  }

  getBookData(book) {
    let { title, authors, imageLinks } = book;
    title = title || "";
    authors = !Array.isArray(authors) ? authors : authors.join(", ");
    const thumbnail =
      imageLinks && _.has(imageLinks, "thumbnail") ? imageLinks.thumbnail : "";
    return { title, authors, thumbnail };
  }
}

Book.propTypes = {
  book: PropTypes.object.isRequired,
  children: PropTypes.node
};

Book.defaultProps = {
  authors: []
};

export default Book;
