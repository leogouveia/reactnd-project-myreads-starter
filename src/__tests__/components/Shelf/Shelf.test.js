import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Shelf } from "../../../components/Shelf/Shelf";
import { Book } from "../../../components/Book/Book";
import { books } from "../../fixtures/fixtures";

it('should render a shelf with books', () => {
  const wrapper = shallow(
      <Shelf name={"Shelf name"}>
        <Book book={books[0]} />
        <Book book={books[1]} />
      </Shelf>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should render a shelf without books', () => {
  const wrapper = shallow(<Shelf name={"Shelf name"} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});