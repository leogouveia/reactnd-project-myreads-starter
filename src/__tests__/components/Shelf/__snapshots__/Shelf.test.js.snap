// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shelf /> should render a shelf with books 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Shelf name
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <Book
        authors={Array []}
        book={
          Object {
            "authors": "Jonh Doe",
            "id": 1,
            "imageLinks": Object {
              "thumbnail": "http://fakeimage.com/image1.png",
            },
            "shelf": "shelf1",
            "title": "Book1 Title",
          }
        }
      />
      <Book
        authors={Array []}
        book={
          Object {
            "authors": "Jane Doe",
            "id": 2,
            "imageLinks": Object {
              "thumbnail": "http://fakeimage.com/image2.png",
            },
            "shelf": "shelf3",
            "title": "Book2 Title",
          }
        }
      />
    </ol>
  </div>
</div>
`;

exports[`<Shelf /> should render a shelf without books 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Shelf name
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    />
  </div>
</div>
`;
